packages/paste-core/components/menu/stories/customization.stories.tsx

```diff
@@ -0,0 +1,231 @@
+    variants: {
+      secondary: {backgroundColor: 'colorBackgroundBusy'},
+      destructive: {backgroundColor: 'colorBackgroundDestructiveDarkest'},
+      destructive_secondary: {backgroundColor: 'colorBackgroundNeutralWeakest', color: 'colorTextWarningStrong'},
+      link: {padding: 'space40', borderRadius: 'borderRadiusCircle', backgroundColor: 'colorBackgroundNeutralWeakest'},
+      destructive_link: {
+        padding: 'space40',
+        borderRadius: 'borderRadiusCircle',
+        backgroundColor: 'colorBackgroundDestructiveLight',
+        color: 'colorTextWarningStrong',
+        fontWeight: 'fontWeightBold',
+      },
+    },
+  },
+  [`SUB${element}_BUTTON`]: {
+    ':hover': {
+      borderLeftWidth: 'borderWidth30',
+      borderLeftStyle: 'solid',
+      color: 'colorTextLink',
+      borderLeftColor: 'colorBorderPrimary',
+      textDecoration: 'none',
+    },
+    ':disabled': {
+      borderLeftWidth: 'borderWidth0',
+    },
+  },
+  [`SUB${element}_MEDIA_FIGURE`]: {
+    ':hover': {
+      color: 'currentColor',
+    },
+  },
+  [`SUB${element}_ICON`]: {
+    paddingLeft: 'space0',
+  },
+  [`${element}_SEPARATOR`]: {
+    borderStyle: 'inset',
+    opacity: 0.5,
+    borderWidth: 'borderWidth40',
+    borderColor: 'colorBorderWeaker',
+  },
+  [`${element}_ITEM`]: {
+    fontWeight: 'fontWeightMedium',
+    ':hover': {
+      borderLeftWidth: 'borderWidth30',
+      borderLeftStyle: 'solid',
+      borderLeftColor: 'colorBorderPrimary',
+      textDecoration: 'none',
+    },
+    ':focus': {
+      borderLeftWidth: 'borderWidth30',
+      borderLeftStyle: 'solid',
+      borderLeftColor: 'colorBorderPrimary',
+      textDecoration: 'none',
+      fontWeight: 'fontWeightBold',
+    },
+    ':disabled': {
+      borderLeftWidth: 'borderWidth0',
+    },
+  },
+});
+
+const getElementName = (elementName: string | undefined, suffix?: string): string | undefined => {
+  const end = suffix ? `_${suffix}` : '';
+  return elementName != null ? `${elementName}${end}` : undefined;
+};
+
+// eslint-disable-next-line import/no-default-export
+export default {
+  title: 'Components/Menu/Customization',
+  component: Menu,
+  subcomponents: {MenuButton, SubMenuButton, MenuGroup, MenuItem, MenuSeparator},
+  parameters: {
+    // Sets a delay for the component's stories
+    chromatic: {delay: 3000},
+  },
+  excludeStories: ['initStyles', 'BaseMenu'],
+};
+
+export const BaseMenu: React.FC<{menuButtonVariant?: ButtonVariants; element?: string}> = React.memo(
+  ({menuButtonVariant = 'primary', element}) => {
+    const menu = useMenuState({
+      visible: true,
+      baseId: `${menuButtonVariant}-menu-customization-story`,
+    });
+
+    const subMenu = useMenuState({baseId: `${menuButtonVariant}-menu-submenu`});
+    const onClick = React.useCallback(() => {
+      menu.hide();
+    }, [menu.hide]);
+    return (
+      <>
+        <MenuButton
+          {...menu}
+          element={getElementName(element, 'BUTTON')}
```

Comment:
Since we're always passing _some_ element here, we're never checking the default. So if someone in the future changed the default value in the component, our tests won't catch it. Do you think we should test for that?

Reply:
This actually does pass the default because it only applies the suffix if element is not undefined. If undefined, it passes undefined, and the component uses its' default.